src/main/kotlin/dev/themeinerlp/attollo/listener/UpdateCheckerListener.kt

```diff
@@ -0,0 +1,33 @@
+package dev.themeinerlp.attollo.listener
+
+import dev.themeinerlp.attollo.Attollo
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.player.PlayerJoinEvent
+import java.net.URI
+
+class UpdateCheckerListener(private val attollo: Attollo) : Listener {
+
+    @EventHandler
+    fun onJoin(event: PlayerJoinEvent) {
+        val player = event.player
+        if (player.hasPermission("attollo.update")) {
+            attollo.server.scheduler.runTaskAsynchronously(attollo, Runnable {
+                val latestVersion = getLatestVersion()
+                if (attollo.description.version != latestVersion) {
+                    player.sendMessage(
+                        "§aAn update for Attollo is available! Version: $latestVersion",
+                        "§aYou can download it at: https://hangar.papermc.io/OneLiteFeather/Attollo"
+                    )
+                }
+            })
```

Comment:
This code is not really resource helpful and starts every time a new scheduler on bukkit. 
As inspiration, please orientate something like that: https://github.com/OneLiteFeatherNET/AntiRedstoneClock-Remastered/blob/develop/src/main/java/net/onelitefeather/antiredstoneclockremastered/service/UpdateService.java

Reply:
Thanks for the feedback. I have just updated the entire system to use one similar to the link you have provided.